pd: add some configuration descriptions (#3538)
pingcap/docs: pd-configuration-file.md

This document only describes parameters that are not included in command-line parameters. Check [here](/command-line-flags-for-pd-configuration.md) for the command line parameters.

### `name`

- The unique name of a PD node
- Default value: `"pd"`
- To start multiply PD nodes, use a unique name for each node.

### `data-dir`

- The directory in which PD stores data
- Default value: `default.${name}"`

### `client-urls`

> - The list of client URLs to be listened to for client traffic

Reviewer: ```suggestion
- The list of client URLs to be listened to by PD
```

- Default value: `"http://127.0.0.1:2379"`

> - To deploy a cluster, you must specify the IP address of the current host as `client-urls` (for example, `"http://192.168.100.113:2379"`). If the cluster runs on Docker, specify the IP address of Docker as `"http://0.0.0.0:2379"`.

Reviewer: ```suggestion
- When you deploy a cluster, you must specify the IP address of the current host as `client-urls` (for example, `"http://192.168.100.113:2379"`). If the cluster runs on Docker, specify the IP address of Docker as `"http://0.0.0.0:2379"`.
```

### `advertise-client-urls`

> - The list of advertise URLs for client traffic from the outside

Reviewer: ```suggestion
- The list of advertise URLs for the client to access PD
```

> - Default value: `"${peer-urls}"`

Reviewer: ```suggestion
- Default value: `"${client-urls}"`
```

> - In some situations such as the Docker or NAT network environment, if a client cannot connect to PD through the default client URLs to be listened to by PD, you must manually set the advertise client URLs.

Reviewer: ```suggestion
- In some situations such as in the Docker or NAT network environment, if a client cannot access PD through the default client URLs listened to by PD, you must manually set the advertise client URLs.
```

- For example, the internal IP address of Docker is `172.17.0.1`, while the IP address of the host is `192.168.100.113` and the port mapping is set to `-p 2380:2380`. In this case, you can set `advertise-client-urls` to `"http://192.168.100.113:2380"`. The client can find this service through `"http://192.168.100.113:2380"`.

### `peer-urls`

> - The list of peer URLs to be listened to for peer traffic

Reviewer: ```suggestion
- The list of peer URLs to be listened to by a PD node
```

- Default value: `"http://127.0.0.1:2380"`

> - To deploy a cluster, you must specify `peer-urls` as the IP address of the current host, such as `"http://192.168.100.113:2380"`. If the cluster runs on Docker, specify the IP address of Docker as `"http://0.0.0.0:2380"`.

Reviewer: ```suggestion
- When you deploy a cluster, you must specify `peer-urls` as the IP address of the current host, such as `"http://192.168.100.113:2380"`. If the cluster runs on Docker, specify the IP address of Docker as `"http://0.0.0.0:2380"`.
```

### `advertise-peer-urls`

> - The list of advertise URLs for peer traffic from the outside

Reviewer: ```suggestion
- The list of advertise URLs for other PD nodes (peers) to access a PD node
```

- Default: `"${peer-urls}"`

> - If the peer cannot connect to PD through the default listening peer URLs, you must manually set the advertise peer URLs explicitly.

Reviewer: ```suggestion
- In some situations such as the Docker or NAT network environment, if the other nodes cannot connect to PD through the default peer URLs to be listened to by PD, you must manually set the advertise peer URLs.
```

> - In some situations such as the Docker or NAT network environment, if the other nodes cannot connect to PD through the default peer URLs to be listened to by PD, you must manually set the advertise peer URLs.

Reviewer: ```suggestion
- In some situations such as in the Docker or NAT network environment, if the other nodes (peers) cannot access the PD node through the default peer URLs listened to by this PD node, you must manually set the advertise peer URLs.
```

> - For example, the internal IP address of Docker is 172.17.0.1, while the IP address of the host is 192.168.100.113 and the port mapping is set to `-p 2380:2380`. In this case, you can set `advertise-peer-urls` to `"http://192.168.100.113:2380"`. The other PD nodes can find this service through `"http://192.168.100.113:2380"`.

Reviewer: ```suggestion
- For example, the internal IP address of Docker is `172.17.0.1`, while the IP address of the host is `192.168.100.113` and the port mapping is set to `-p 2380:2380`. In this case, you can set `advertise-peer-urls` to `"http://192.168.100.113:2380"`. The other PD nodes can find this service through `"http://192.168.100.113:2380"`.
```

### `initial-cluster`

- The initial cluster configuration for bootstrapping
- Default value: `"{name}=http://{advertise-peer-url}"`
- For example, if `name` is "pd", and `advertise-peer-urls` is `"http://192.168.100.113:2380"`, the `initial-cluster` is `"pd=http://192.168.100.113:2380"`.
- If you need to start three PD servers, the `initial-cluster` might be:

    ```
    pd1=http://192.168.100.113:2380, pd2=http://192.168.100.114:2380, pd3=192.168.100.115:2380
    ```

### `initial-cluster-state`

+ The initial state of the cluster
+ Default value: `"new"`

### `initial-cluster-token`

> + Used to identify different clusters during the bootstrap phase.

Reviewer: ```suggestion
+ Identifies different clusters during the bootstrap phase.
```

> + Default value: "pd-cluster"

Reviewer: ```suggestion
+ Default value: `"pd-cluster"`
```

> + If multiple clusters that have nodes with same configurations are deployed successively, different tokens should be specified to isolate different cluster nodes.

Reviewer: ```suggestion
+ If multiple clusters that have nodes with same configurations are deployed successively, you must specify different tokens to isolate different cluster nodes.
```
